index.js

```diff
@@ -499,10 +499,7 @@ const legacyScriptingRunner = (Zap, zcli, input) => {
   };
 
   const runOAuth1GetRequestToken = async bundle => {
-    const url = _.get(
-      app,
-      'authentication.oauth1Config.legacyProperties.requestTokenUrl'
-    );
+    const url = _.get(app, 'legacy.authentication.requestTokenUrl');
```

Comment:
One thing to get some frontend perspective on is how paths like these will work with saving data to the backend. For example, if we show this field inline in the UI, would it help to have the path be `legacy.authentication.oauth1Config.requestTokenUrl`, which mirrors the regular path, just prepends `legacy`.

@stevelikesmusic Curious if you have thoughts on this. For a bit of context, we want to swap up the schema so that an app that has been converted from the existing WB to the new one will have all the "legacy" properties unique to conversion under a single key on the app. Compare this to today, where all over the schema we have a `legacyProperties`.

So my question is, if we had a component that needed to save data to two spots within the app definition, does it help to keep the path consistent, except one location gets prefixed with `legacy`. Basically:

```
fetch('/api/app/3', {
 "legacy": {
    "authentication": {
        ... <------ does it help frontend code if structure here matches below
    }
  },
    "authentication": {
      ...
    }
})
```

Reply:
My initial thought is, yes throw it all behind `legacy` with the same access pattern as the v3 schema when possible. I don't _quite_ understand how converted apps will work within Godzilla or the v3 platform in regards to legacy functionality from WB. For instance, will a converted app require pieces from both `app.authentication` and `app.legacy.authentication`? Will a converted app's Godzilla view be the same as an app started with Godzilla?

I think it will for sure make things easier to understand if legacy components/props are mapped under `legacy`. And then accessing or mutating those props is as simple as prepending `legacy` like you say @bcooksey . 

Do I understand correctly @eliangcs that `legacy` props on the app-definition would only be accessed within the Advanced tab in Godzilla? We would expect everything else to adhere to the v3 schema/platform? So saving a converted app's trigger would only affect `app.triggers.<triggerKey>` when editing in Godzilla?

Or, for example, would someone with a converted app be able to make changes to auth that affected `app.authentication` _and_ `app.legacy.authentication` from the Godzilla Authentication section? Meaning, we would have additional form fields or options to show in Auth for converted apps that we don't current apps started within Godzilla.

Depending on how encapsulated or widespread legacy components will be within Godzilla, we can always start separating logic with handlers. Then working with the app definition becomes much easier, though this would be overkill if legacy logic is kept behind a single tab (Advanced) in Godzilla.

Reply:
@stevelikesmusic those are great questions! I'll answer them as follows. @ZaneLyon @bcooksey feel free to correct me if you found anything inaccurate.

> will a converted app require pieces from both app.authentication and app.legacy.authentication?

Yes, a converted app will require both. Things that are already in the v3 schema, such as `app.authentication` and `app.triggers`, will be present for converted apps, too. But we'll fill in a shim code that invokes legacy-scripting-runner (i.e. this repo) for those fields. For example, take a look at `triggers.mailbox.operation.perform` in https://cdn.zapier.com/storage/files/ccaed092b510d089adb863e5bbf9c3a3.json. The shim code `return z.legacyScripting.run(bundle, 'trigger', 'mailbox');` invokes legacy-scripting-runner, which then uses properties in `app.legacy` to simulate the legacy Web Builder.

> Will a converted app's Godzilla view be the same as an app started with Godzilla?

They will look alike except for in the Advanced tab.

> legacy props on the app-definition would only be accessed within the Advanced tab in Godzilla?

Yes.

> We would expect everything else to adhere to the v3 schema/platform? So saving a converted app's trigger would only affect `app.triggers.<triggerKey>` when editing in Godzilla?

Yes.

> would someone with a converted app be able to make changes to auth that affected `app.authentication` _and_ `app.legacy.authentication` from the Godzilla Authentication section?

No, they can only affect `app.authentication` from the Godzilla Authentication section.

Reply:
Thanks explaining @eliangcs 👏 👏 👏! You gave me a some much needed clarity with converted apps!

In that case, I would say definitely keep the same access pattern when possible. It will indeed simplify how we interact with `legacy` props and hopefully make it easier for newcomers to understand the relationships between `legacy` components from WB with current Godzilla paradigms.

Reply:
> legacy props on the app-definition would only be accessed within the Advanced tab in Godzilla?

We can decide this based on whats best UX and whats easy/hard technically. I can think of a case where it would be a better UX to expose a prop in another part of the Godzilla UI. For example, a RESTHook trigger can have a polling URL and a hookType in a converted app, but not in a v3 schema app. From UX perspective, would it make most sense to put those props in where you edit a trigger, or on Advanced tab and make devs go to two places to control a trigger? To me seems like combining into one is best there, unless it makes the code really complicated.

Other things though, like auth mapping or legacy scripting, would be on advanced tab.

Reply:
For posterity, on a team call we discussed and decided we'd have a single tab where you edit all the legacy stuff.